astropy/time/core.py

```diff
@@ -3237,30 +3313,43 @@ def get_as_filled_ndarray(mask, val):
 
         Note that nothing happens if there are no masked elements.
         """
-        fill_value = None
+        if isinstance(val, np.ma.MaskedArray):
```

Comment:
> I think there is a docstring mistake above. It says that the unmasked case returns `mask=None` but it should be `False`. I noticed this in the code that uses this function since it checks for `mask is not False`.

I think you fixed this already yourself, so resolving...